app/controllers/sessions_controller.rb

```diff
@@ -1,30 +1,67 @@
 class SessionsController < ApplicationController
+  skip_before_action :require_login, only: [:create]
+
   def login_form
   end
 
-  def login
-    username = params[:username]
-    if username and user = User.find_by(username: username)
-      session[:user_id] = user.id
-      flash[:status] = :success
-      flash[:result_text] = "Successfully logged in as existing user #{user.username}"
+  def create
+    auth_hash = request.env['omniauth.auth']
+
+    @user = User.find_by(uid: auth_hash[:uid], provider: 'github')
+
+    if @user
+      session[:user_id] = @user.id
+      flash[:success] = "Welcome #{@user.username}"
+
+      # if @user.nil?
+        # User doesn't match anything in the DB
+        # Attempt to create a new user
+
+
+      # else
+
+        # flash[:success] = "Logged in successfully"
+        # redirect_to root_path
+      # end
     else
-      user = User.new(username: username)
-      if user.save
-        session[:user_id] = user.id
-        flash[:status] = :success
-        flash[:result_text] = "Successfully created new user #{user.username} with ID #{user.id}"
+      @user = User.new(uid: auth_hash['uid'], provider: auth_hash['provider'], username: auth_hash['info']['nickname'], email: auth_hash['info']['email'])
+
```

Comment:
Pulling all this data out of the `auth_hash` is complex. This would be a good candidate for a model method, something like `User.from_auth_hash`.